src/ops.rs

```diff
@@ -201,3 +201,48 @@ impl_op_unary! { impl Neg, neg for  Matrix<T, M, N> }
 impl_op_unary! { impl Neg, neg for &Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for  Matrix<T, M, N> }
 impl_op_unary! { impl Not, not for &Matrix<T, M, N> }
+
+////////////////////////////////////////////////////////////////////////////////
+// Matrix * Matrix
+////////////////////////////////////////////////////////////////////////////////
+
+impl<T, const N: usize, const M: usize> Matrix<T, N, M> {
+    pub(crate) fn matmul_into<const P: usize>(
+        &self,
+        rhs: &Matrix<T, M, P>,
+        out: &mut Matrix<T, N, P>,
+    ) where
+        T: Copy + Default + Mul<Output = T> + Add<Output = T>,
+    {
+        for (j, out) in out.data.iter_mut().enumerate() {
+            for (i, out) in out.iter_mut().enumerate() {
+                let mut acc = T::default();
+                for k in 0..M {
+                    acc = acc + self[(i, k)] * rhs[(k, j)];
+                }
+                *out = acc;
```

Comment:
You should probably use `T::zero()` here, it is more clear. You could also use `Sum` and you won't have to call `T::default()`.

```rust
            *out = (0..M).map(|k| self[(i, k)] * rhs[(k, j)]).sum()
```

Obviously this requires `Sum` now, but its pretty commonly implemented.

Reply:
Product sum is now done this way, good idea!